### DISCUSSION TITLE
Connector builder tools!

### DISCUSSION THREAD

**Discussion by Laskewitz** [microsoft/PowerPlatformConnectors#1416] [Building Connectors]
Let's start a list of useful tools that you can use for building your own custom / certified / independent publisher connectors. I'll keep updating the list of tools when suggestions in this discussion appear. 

### **Tools**

- [Visual Studio Code](https://code.visualstudio.com/)
- [Paconn CLI](https://docs.microsoft.com/en-us/connectors/custom-connectors/paconn-cli)
- [Windows Terminal](https://aka.ms/terminal)
- [Postman](https://www.postman.com/)
- [GitHub Copilot](https://copilot.github.com/)

### NPM Packages
- [API Spec Converter](https://github.com/LucyBot-Inc/api-spec-converter) (added by @troystaylor)
- [Swagger-Markdown](https://www.npmjs.com/package/swagger-markdown) (added by @troystaylor)

### **Visual Studio Code Extensions**

- [Power Platform Connectors](https://aka.ms/ppc-vscode)
- [OpenAPI (Swagger) Editor](https://marketplace.visualstudio.com/items?itemName=42Crunch.vscode-openapi) (added by @troystaylor)
- [Error Lens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens)

**Comment by troystaylor** [comment]
My toolbox is:

- Visual Studio Code
- [swagger-markdown](https://www.npmjs.com/package/swagger-markdown)
- [Atom editor](https://atom.io/) - creating readme.md file from swagger-markdown file strings (great to also check spelling for apiDefinition.swagger.json file)
- [API Spec Converter](https://github.com/LucyBot-Inc/api-spec-converter) - not perfect but helps to convert OpenAPI 3.0 definition file to Swagger

**Comment by Hjaf** [comment]
What about [spectral](https://meta.stoplight.io/docs/spectral/674b27b261c3c-overview)? 
A set of rules that can be used to validate OAS, and help with highlighting the changes needed for definitions to be used in a custom connector would be of great help. 

There is a [vscode extension](https://marketplace.visualstudio.com/items?itemName=stoplight.spectral) as well, this can also be useful for pipelines that maintains custom connectors
